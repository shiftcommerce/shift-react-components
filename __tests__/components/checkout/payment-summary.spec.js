// Libraries
import React from 'react'

// Components
import PaymentSummary from '../../../src/components/checkout/payment-summary'

// Objects
import Flash from '../../../src/objects/flash'

test("doesn't add errors class when there are no errors", () => {
  const wrapper = shallow(<PaymentSummary headerTitle={'Paymment'} billingAddress={{}} />)

  expect(wrapper.find('div')).not.toMatchSelector('.o-form__error')
})

test('add errors class when errors are present', () => {
  const wrapper = shallow(<PaymentSummary headerTitle={'Paymment'} billingAddress={{}} withErrors={true} />)

  expect(wrapper.find('div')).toMatchSelector('.o-form__error')
})

test('renders error messages', () => {
  // Arrange
  const errorMessage = 'Something went wrong'
  const headerTitle = 'Payment Details'

  // Act
  const wrapper = shallow(
    <PaymentSummary
      headerTitle={headerTitle}
      billingAddress={{}}
      errorMessage={errorMessage}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find(Flash).length).toEqual(1)
})
