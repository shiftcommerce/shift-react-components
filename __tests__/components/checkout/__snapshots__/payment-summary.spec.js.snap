// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error messages 1`] = `
<Fragment>
  <PaymentHeader
    collapsed={true}
    title="Payment Details"
  />
  <div
    className="c-payment__summary"
  >
    <_default
      modifier="error"
      text="Something went wrong"
    />
    <p>
      <span
        className="u-bold"
      >
        Billing Address: 
      </span>
      <span
        className="u-bold"
      >
         
         
      </span>
      <span>
        , 
        , 
      </span>
    </p>
  </div>
</Fragment>
`;
